### PR TITLE
Add whitespace removal inside template curly brackes

### PR DESCRIPTION
**Pull request by silverwind**
Remove spaces inside template curly brackets and add it to `make fmt` and verify in `make fmt-check`.

The `sed` expressions work on both GNU and BSD variants and for that reason, they are using pretty ancient regexp syntax (BSD sed can't do `\s`, `\t`, `+`). The closing bracket expression has an exception for allow multi-line content so, `}}` on new line won't get dedented.

### CODE REVIEW

**Review by Gusted — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -242,8 +242,9 @@ clean:
 
 .PHONY: fmt
 fmt:
-	@echo "Running gitea-fmt (with gofumpt)..."
 	@MISSPELL_PACKAGE=$(MISSPELL_PACKAGE) GOFUMPT_PACKAGE=$(GOFUMPT_PACKAGE) $(GO) run build/code-batch-process.go gitea-fmt -w '{file-list}'
+	$(eval TEMPLATES := $(wildcard templates/**/*.tmpl))
+	@$(SED_INPLACE) -e 's/{{[ 	]\{1,\}/{{/g' -e '/^[ 	]\{1,\}}}/! s/[ 	]\{1,\}}}/}}/g' $(TEMPLATES)
```
Comment (delvh) [inline Makefile:248]: Is it only me, or is the second expression weird?
I can completely understand the first one, but the second seems absolutely wrong:
Why doesn't it start with `s/`?
Why is the `s/` in the middle?
What's the `/! s/`?
And am I right to assume that it is some kind of unnecessary condition?

Comment (silverwind) [reply Makefile:248]: I picked it up from [here](https://stackoverflow.com/a/12178023/808699), it's basically a "ignore pattern" that makes the subsequent replacement not apply. It is for cases like this:

```handlebars
  {{
    stuff
  }}
```

without it, the `}}` would get dedented, which is not desirable.

Comment (silverwind) [inline Makefile:248]:
```suggestion
	@# strip whitespace after '{{' and before `}}` unless there is only whitespace before it 
	@$(SED_INPLACE) -e 's/{{[ 	]\{1,\}/{{/g' -e '/^[ 	]\{1,\}}}/! s/[ 	]\{1,\}}}/}}/g' $(TEMPLATES)
```

Comment (silverwind) [reply Makefile:248]: Added a comment describing it. Rewriting it in `perl` might result in a much more readable regexp, but I'm not sure whether we can depend on it (likely can, thought).

Comment (silverwind) [reply Makefile:248]: I think sed is fine for now. If we are to switch to perl, we should replace all `sed` calls with it.

**Pull request merged**